pkg/container-disk/container-disk_test.go

```diff
@@ -346,43 +392,59 @@ var _ = Describe("ContainerDisk", func() {
 		})
 
 		Context("should use the right containerID", func() {
+
 			It("for a new migration pod with two containerDisks", func() {
 				clusterConfig, _, _ := testutils.NewFakeClusterConfigUsingKVConfig(&v1.KubeVirtConfiguration{
 					SupportContainerResources: []v1.SupportContainerResources{},
 				})
-				vmi := api.NewMinimalVMI("myvmi")
-				appendContainerDisk(vmi, "disk1")
-				appendNonContainerDisk(vmi, "disk3")
-				appendContainerDisk(vmi, "disk2")
+
+				By("Creating a new VMI with container and non-container disks")
+				vmi := libvmi.New(
+					libvmi.WithContainerDisk("disk1", "someimage:v1.2.3.4"),
+					libvmi.WithDataVolume("disk3", "some-data-volume"),
+					libvmi.WithContainerDisk("disk2", "someimage:v1.2.3.4"),
+				)
 
 				pod := createMigrationSourcePod(vmi)
 
+				By("Extracting image IDs from the source pod")
 				imageIDs := ExtractImageIDsFromSourcePod(vmi, pod)
 				Expect(imageIDs).To(HaveKeyWithValue("disk1", "someimage@sha256:0"))
 				Expect(imageIDs).To(HaveKeyWithValue("disk2", "someimage@sha256:1"))
 				Expect(imageIDs).To(HaveLen(2))
 
+				By("Generating new containers based on the extracted image IDs")
 				newContainers := GenerateContainers(vmi, clusterConfig, imageIDs, "a-name", "something")
 				Expect(newContainers[0].Image).To(Equal("someimage@sha256:0"))
 				Expect(newContainers[1].Image).To(Equal("someimage@sha256:1"))
 			})
+
 			It("for a new migration pod with a containerDisk and a kernel image", func() {
 				clusterConfig, _, _ := testutils.NewFakeClusterConfigUsingKVConfig(&v1.KubeVirtConfiguration{
 					SupportContainerResources: []v1.SupportContainerResources{},
 				})
-				vmi := api.NewMinimalVMI("myvmi")
-				appendContainerDisk(vmi, "disk1")
-				appendNonContainerDisk(vmi, "disk3")
 
-				vmi.Spec.Domain.Firmware = &v1.Firmware{KernelBoot: &v1.KernelBoot{Container: &v1.KernelBootContainer{Image: "someimage:v1.2.3.4"}}}
+				By("Creating a new VMI with a container disk and a kernel boot image")
+				vmi := libvmi.New(
+					libvmi.WithContainerDisk("disk1", "someimage:v1.2.3.4"),
+					libvmi.WithDataVolume("disk3", "some-data-volume"),
+				)
+				vmi.Spec.Domain.Firmware = &v1.Firmware{
+					KernelBoot: &v1.KernelBoot{
+						Container: &v1.KernelBootContainer{Image: "someimage:v1.2.3.4"},
```

Comment:
Is there a `libvmi` `Option` for this?

Reply:
if you mean for Firmware 
I tried to find the option but I couldn't
The only way was to set it explicitly

Reply:
We should create the option in libvmi, you can look at some examples in `pkg/libvmi`

Reply:
okay i am really exciting to do that
it maybe better to complete this task first and than i will create options for `Firmware` and `WithPullPolicy` and i will back here and update these fields 
so what do you think ?

Reply:
You could first create the 2 options in 2 separate commits and add them on top of this PR and then modify the tests accordingly.

Reply:
I feel i will mess up with commits and branches , Is it okay if did it one by one
fixing the task in hand and then implement the options and just get back and replace it will be much easy 
and i can make sure my implementation is correct

Reply:
I think it is a good opportunity to learn git. I would suggest you to take some tutorials, then it becomes a common work flow.